connector/connect/client/jvm/pom.xml

```diff
@@ -112,40 +124,57 @@
           <shadedArtifactAttached>false</shadedArtifactAttached>
           <artifactSet>
             <includes>
+              <include>com.google.android:*</include>
+              <include>com.google.api.grpc:*</include>
+              <include>com.google.code.findbugs:*</include>
+              <include>com.google.code.gson:*</include>
+              <include>com.google.errorprone:*</include>
               <include>com.google.guava:*</include>
-              <include>io.grpc:*</include>
+              <include>com.google.j2objc:*</include>
               <include>com.google.protobuf:*</include>
+              <include>io.grpc:*</include>
+              <include>io.netty:*</include>
+              <include>io.perfmark:*</include>
+              <include>org.codehaus.mojo:*</include>
+              <include>org.checkerframework:*</include>
               <include>org.apache.spark:spark-connect-common_${scala.binary.version}</include>
             </includes>
           </artifactSet>
           <relocations>
             <relocation>
               <pattern>io.grpc</pattern>
-              <shadedPattern>${spark.shade.packageName}.connect.client.grpc</shadedPattern>
+              <shadedPattern>${spark.shade.packageName}.connect.client.io.grpc</shadedPattern>
               <includes>
                 <include>io.grpc.**</include>
               </includes>
             </relocation>
             <relocation>
-              <pattern>com.google.protobuf</pattern>
-              <shadedPattern>${spark.shade.packageName}.connect.protobuf</shadedPattern>
-              <includes>
-                <include>com.google.protobuf.**</include>
-              </includes>
+              <pattern>com.google</pattern>
+              <shadedPattern>${spark.shade.packageName}.connect.client.com.google</shadedPattern>
             </relocation>
             <relocation>
-              <pattern>com.google.common</pattern>
-              <shadedPattern>${spark.shade.packageName}.connect.client.guava</shadedPattern>
-              <includes>
-                <include>com.google.common.**</include>
-              </includes>
+              <pattern>io.netty</pattern>
```

Comment:
if we shade and relocation netty, the content in `META-INF/native-image/io.netty` should also relocated, just as in `grpc-netty-shaded`.

Now, the content of `META-INF/native-image/io.netty/netty-codec-http2/native-image.properties` is

```
Args = --initialize-at-build-time=io.netty \
       --initialize-at-run-time=io.netty.handler.codec.http2.Http2CodecUtil,io.netty.handler.codec.http2.Http2ClientUpgradeCodec,io.netty.handler.codec.http2.Http2ConnectionHandler,io.netty.handler.codec.http2.DefaultHttp2FrameWriter
```

but I think it should like 

```
Args = --initialize-at-build-time=org.sparkproject.connect.client.io.netty \
       --initialize-at-run-time=org.sparkproject.connect.client.io.netty.handler.codec.http2.Http2CodecUtil,org.sparkproject.connect.client.io.netty.handler.codec.http2.Http2ClientUpgradeCodec,org.sparkproject.connect.client.io.netty.handler.codec.http2.Http2ConnectionHandler,org.sparkproject.connect.client.io.netty.handler.codec.http2.DefaultHttp2FrameWriter       
```
Other Transformer may need to be added

Reply:
https://github.com/grpc/grpc-java/blob/61f19d707a4c6fd4f5e327b8fceb09658b9cb452/netty/shaded/build.gradle#L139-L176

Reply:
I suggest not to shade netty related content

Reply:
Let me come back to you. grpc has this grpc with shaded netty and the one without. We used the one without netty.

Reply:
Yes,  you are right.

To clarify, I mean we need to ensure that the `native-image.properties` files in `META-INF/native-image/io.netty/netty-codec-xxx/` are rewritten correctly if we need to shade netty

Reply:
I understand this file is for graal support. If so, can we leave it in another PR? I do not have any good idea how to use graal and verify it works as expected.

Reply:
No, we should make sure the result of maven shade is correct, because Spark uses maven to make distribution and deploy jar to maven central repository now, not sbt. If there is no clear solution now, I think we should not shade Netty temporarily.

Reply:
The maven shading do not require the file `native-image.propertie`. It is a graal file. Isn't? For a normal maven project point of view, the current solution is already complete. See [maven shading examples](https://maven.apache.org/plugins/maven-shade-plugin/examples/class-relocation.html). Or did I missed anything here?

Reply:
OK,  I see.

Reply:
Please wait for me to do some more check, will feedback tomorrow, thanks @zhenlineo